get-cabal-flags.hs

```diff
@@ -0,0 +1,8 @@
+import Data.Version (Version (Version))
+import Distribution.Simple.Utils (cabalVersion)
+
+main :: IO ()
+main =
+  putStrLn $ if cabalVersion >= Version [1,22] []
+             then "(\"-DuseCompilerInfo\")"
+             else "()"
```

Comment:
Do we need another Haskell Helper for this?  Why don't we just call `cabal --version` from Emacs Lisp to check the Cabal version?  I think we can just assume that cabal-install is present, can't we?